
Triggers on resume ..

[  487.909349] WARNING: CPU: 0 PID: 2125 at drivers/net/wireless/ath/ath10k/mac.c:5625 ath10k_bss_info_changed+0xb33/0xd90 [ath10k_core]
[  487.909350] Modules linked in: ctr ccm cmac af_packet arc4 xt_tcpudp xt_state xt_conntrack nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 libcrc32c bnep iptable_filter ip_tables x_tables bpfilter nls_utf8 nls_cp437 vfat fat amdgpu chash gpu_sched joydev uvcvideo iTCO_wdt snd_hda_codec_hdmi iTCO_vendor_support videobuf2_vmalloc videobuf2_memops videobuf2_v4l2 snd_hda_codec_conexant intel_wmi_thunderbolt videobuf2_common btusb snd_hda_codec_generic radeon btrtl btbcm btintel ath10k_pci videodev bluetooth ath10k_core media i915 ttm ecdh_generic coretemp ath intel_rapl x86_pkg_temp_thermal rtsx_usb_ms intel_powerclamp mac80211 memstick kvm_intel kvmgt vfio_mdev mdev vfio_iommu_type1 vfio kvm snd_soc_skl snd_soc_skl_ipc snd_soc_sst_ipc snd_soc_sst_dsp snd_hda_ext_core snd_soc_acpi_intel_match snd_soc_acpi
[  487.909390]  cec snd_soc_core rc_core drm_kms_helper efi_pstore irqbypass cfg80211 snd_compress drm snd_pcm_dmaengine psmouse intel_gtt evdev snd_hda_intel mac_hid agpgart i2c_algo_bit snd_hda_codec r8169 pcspkr efivars fb_sys_fops syscopyarea sysfillrect sysimgblt i2c_i801 snd_hda_core hwmon i2c_core libphy snd_hwdep ideapad_laptop sparse_keymap intel_pch_thermal rfkill thermal wmi battery acpi_pad ac video pcc_cpufreq button ppdev sch_fq_codel fuse snd_pcm_oss snd_mixer_oss snd_pcm snd_seq_dummy snd_seq_oss snd_seq_midi_event snd_seq snd_seq_device snd_timer snd soundcore lp parport_pc parport binfmt_misc sg ext4 crc32c_generic crc16 mbcache jbd2 fscrypto rtsx_usb_sdmmc mmc_core rtsx_usb sr_mod sd_mod cdrom crct10dif_pclmul crc32_pclmul crc32c_intel ghash_clmulni_intel pcbc xhci_pci ahci xhci_hcd
[  487.909436]  libahci libata aesni_intel aes_x86_64 crypto_simd usbcore cryptd scsi_mod glue_helper serio_raw dm_mirror dm_region_hash dm_log dm_mod efivarfs unix sha256_mb sha256_ssse3 sha256_generic sha1_mb mcryptd sha1_ssse3 sha1_generic hmac ipv6 autofs4
[  487.909454] CPU: 0 PID: 2125 Comm: kworker/u8:21 Not tainted 4.19.2-fw1 #1
[  487.909455] Hardware name: LENOVO 80UD/LNVNB161216, BIOS 1TCN26WW(V2.07) 03/29/2018
[  487.909459] Workqueue: events_unbound async_run_entry_fn
[  487.909466] RIP: 0010:ath10k_bss_info_changed+0xb33/0xd90 [ath10k_core]
[  487.909468] Code: ff ff b8 a1 ff ff ff e9 e9 f7 ff ff b8 a1 ff ff ff e9 a5 f6 ff ff b8 a1 ff ff ff e9 ef f5 ff ff b8 a1 ff ff ff e9 4f f7 ff ff <0f> 0b e9 37 f8 ff ff b8 a1 ff ff ff e9 f6 fa ff ff b8 a1 ff ff ff
[  487.909469] RSP: 0018:ffff9b9042e47cd8 EFLAGS: 00010282
[  487.909471] RAX: 00000000fffffffe RBX: ffff904af2fb1540 RCX: 0000000000000000
[  487.909472] RDX: ffff904af2fb18b8 RSI: ffff9b9042e47cf8 RDI: ffff904af4181380
[  487.909473] RBP: ffff904af4181380 R08: 0000000000200000 R09: 0000000000000000
[  487.909474] R10: 000000000000001f R11: ffff904ae2b4a600 R12: 0000000002000000
[  487.909475] R13: ffff904af4181388 R14: ffff904af2fb24d0 R15: ffff904af2fb1540
[  487.909477] FS:  0000000000000000(0000) GS:ffff904af7200000(0000) knlGS:0000000000000000
[  487.909478] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[  487.909479] CR2: 00007fa6a8300510 CR3: 000000002000a002 CR4: 00000000003606f0
[  487.909480] Call Trace:
[  487.909488]  ? ath10k_conf_tx+0x12d/0x4c0 [ath10k_core]
[  487.909503]  ieee80211_bss_info_change_notify+0xa9/0x1c0 [mac80211]
[  487.909521]  ieee80211_reconfig+0x9d7/0x14f0 [mac80211]
[  487.909538]  wiphy_resume+0x7e/0x150 [cfg80211]
[  487.909549]  ? wiphy_namespace+0x10/0x10 [cfg80211]
[  487.909553]  dpm_run_callback+0x2e/0x130
[  487.909556]  device_resume+0x97/0x190
[  487.909558]  async_resume+0x19/0x40
[  487.909561]  async_run_entry_fn+0x37/0xe0
[  487.909563]  process_one_work+0x1e9/0x410
[  487.909565]  worker_thread+0x2d/0x3d0
[  487.909567]  ? process_one_work+0x410/0x410
[  487.909569]  kthread+0x113/0x130
[  487.909572]  ? kthread_park+0x90/0x90
[  487.909575]  ret_from_fork+0x35/0x40
[  487.909577] ---[ end trace 56a3ea97193bc4c5 ]---

Introduced by:

.....

commit cd93b83ad927b2c7979e0add0343ace59328b461
Author: Pradeep Kumar Chitrapu <pradeepc@codeaurora.org>
Date:   Wed Jul 25 10:59:39 2018 +0300

    ath10k: support for multicast rate control
    
    Issues a wmi command to firmware when multicast rate change is received with the
    new BSS_CHANGED_MCAST_RATE flag.  Also fixes the incorrect fixed_rate setting
    for CCK rates which got introduced with addition of ath10k_rates_rev2 enum.
    
    Tested on QCA9984 with firmware ver 10.4-3.6-00104
    
    Signed-off-by: Pradeep Kumar Chitrapu <pradeepc@codeaurora.org>
    Signed-off-by: Kalle Valo <kvalo@codeaurora.org>

.....


diff -Naur linux-4.19.2/drivers/net/wireless/ath/ath10k/mac.c linux-4.19.2-p/drivers/net/wireless/ath/ath10k/mac.c
--- linux-4.19.2/drivers/net/wireless/ath/ath10k/mac.c	2018-11-13 20:09:00.000000000 +0100
+++ linux-4.19.2-p/drivers/net/wireless/ath/ath10k/mac.c	2018-11-15 01:41:51.896601274 +0100
@@ -5621,8 +5621,7 @@
 				    arvif->vdev_id, ret);
 	}
 
-	if (changed & BSS_CHANGED_MCAST_RATE &&
-	    !WARN_ON(ath10k_mac_vif_chan(arvif->vif, &def))) {
+	if (changed & BSS_CHANGED_MCAST_RATE && !ath10k_mac_vif_chan(arvif->vif, &def)) {
 		band = def.chan->band;
 		rateidx = vif->bss_conf.mcast_rate[band] - 1;
 
